File: JSPatchDemo/JPViewController.m
//
//  JPViewController.m
//  JSPatchDemo
//
//  Created by 马栋军 on 2018/8/15.
//  Copyright © 2018年 DangDangWang. All rights reserved.
//

#import "JPViewController.h"
#import "JPTableViewController.h"

@interface JPViewController ()

@end

@implementation JPViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    UIButton *btn = [[UIButton alloc] initWithFrame:CGRectMake(0, 100, [UIScreen mainScreen].bounds.size.width, 50)];
    [btn setTitle:@"Push JPTableViewController" forState:UIControlStateNormal];
    [btn addTarget:self action:@selector(handleBtn:) forControlEvents:UIControlEventTouchUpInside];
    [btn setBackgroundColor:[UIColor grayColor]];
    [self.view addSubview:btn];
}

- (void)handleBtn:(id)sender
{
    UIViewController *vc = [[UIViewController alloc] init];
    vc.title = @"UIViewController";
    [self.navigationController pushViewController:vc animated:YES];
}

@end

File: JSPatchDemo/JPViewController.js
// 测试 handleBtn
require('JPTableViewController,UIColor')
defineClass('JPViewController',{
            handleBtn: function(sender){
            var vc = JPTableViewController.alloc().init();
            vc.setTitle("JPViewController.js-handleBtn");
            var view = vc.view();
            var bgColor = UIColor.redColor();
            view.setBackgroundColor(bgColor);
            
        self.navigationController().pushViewController_animated(vc,YES);
            }
            })
